core/lib/src/response/named_file.rs

```diff
@@ -80,13 +105,30 @@ impl NamedFile {
 /// implied by its extension, use a [`File`] directly.
 impl Responder<'_> for NamedFile {
     fn respond_to(self, req: &Request<'_>) -> response::Result<'static> {
-        let mut response = self.1.respond_to(req)?;
-        if let Some(ext) = self.0.extension() {
+        let last_modified = self
+            .modified
+            .map(|modified| modified.format("%a, %d %b %Y %T GMT"));
+
+        if let Some(if_modified_since) = req.headers().get("If-Modified-Since").next() {
+            if let Some(last_modified) = &last_modified {
+                if last_modified == if_modified_since {
+                    return Response::build().status(Status::NotModified).ok();
+                }
+            }
+        }
+
+        let mut response = self.file.respond_to(req)?;
+        if let Some(ext) = self.path.extension() {
             if let Some(ct) = ContentType::from_extension(&ext.to_string_lossy()) {
                 response.set_header(ct);
             }
         }
 
+        if let Some(last_modified) = last_modified {
+            response.set_header(Header::new("Last-Modified", last_modified));
+            response.set_header(Header::new("Cache-Control", "public, max-age=31536000"));
```

Comment:
I'm not 100% sure if this can be omitted but it seems that it is not necessary. According to RFC7232 the server must send it in 304 when it has been send in the 200 response. 

> The server generating a 304 response MUST generate any of the
> following header fields that would have been sent in a 200 (OK)
> response to the same request: Cache-Control, Content-Location, Date,
> ETag, Expires, and Vary.

I tested it with Firefox and it worked.

Reply:
I looked at the definitions again, and I think that setting `Cache-Control` at all is too opinionated of a decision to do directly in `NamedFile`. This particular value states the content is not expected to change for a year, after which point the browser should re-request it.

Reply:
It has been removed.